calc/src/mechanics/gen789.ts

```diff
@@ -813,10 +813,12 @@ export function calculateBasePowerSMSSSV(
   basePower = OF16(Math.max(1, pokeRound((basePower * chainMods(bpMods, 41, 2097152)) / 4096)));
   if (
     attacker.teraType && move.type === attacker.teraType &&
-    attacker.hasType(attacker.teraType) && !move.hits &&
-    move.priority <= 0 && basePower <= 60 && gen.num >= 9
+    attacker.hasType(attacker.teraType) && move.hits === 1 &&
+    move.priority <= 0 && move.bp > 0 && !move.named('Dragon Energy', 'Eruption', 'Water Spout') &&
```

Comment:
The `move.bp > 0` check is there so that the ineligible moves do not receive a boost. These moves (aside from Dragon Energy, Eruption, and Water Spout) have variable base power (Gyro Ball, Electro Ball, Heavy Slam, Reversal, etc.), so in the move data they will have a `bp` of 0 (though the `basePower` variable will be more than 0 for a damaging move). Without this check, these moves would be upped to 60 BP by Tera in situations where they would normally have less than 60 BP.

With the check:
`0 Atk Tera Steel Electrode Gyro Ball (6 BP) vs. 0 HP / 0 Def Ferrothorn: 2-3 (0.6 - 1%) -- possibly the worst move ever`
Without the check:
`0 Atk Tera Steel Electrode Gyro Ball (60 BP) vs. 0 HP / 0 Def Ferrothorn: 15-18 (5.1 - 6.2%) -- possibly the worst move ever`

Reply:
Makes sense, thanks!